### PR TITLE
CarApplication

### PR DESCRIPTION
**Pull request by MehmedShehov**
First Task

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/*.class
```
Comment (vanyamihova-deltasource) [inline .gitignore:1]: There are some templates of `.gitignore`. Better to find some and use it

File: CarApp/CarApp.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nikolayivanov-deltasource) [inline CarApp/CarApp.iml:1]: Add this file to .gitignore. It's created by IntelliJ

File: CarApp/src/application/Car.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+/**
+ * DeltaSource Internship - first task
+ * The class represents the specs of a car
+ * the weight is in kg and consumption in litres
+ *
+ * @author Mehmed Shehov
+ */
+public class Car {
+    final static int maxWeight = 3500;
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Car.java:11]: make private and UPPER_CASE

File: CarApp/src/application/Car.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    enum SteeringWheel {
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Car.java:13]: Better move the enums in separate files

File: CarApp/src/application/Application.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package application;
+
+/**
+ * DeltaSource Internship - first task
+ * This  class is used to test and print the specs of the Car class
+ *
+ * @author - Mehmed Shehov
+ */
+
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Application.java:9]: You don't need those blank lines

File: CarApp/src/application/Car.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    private String color;
+    private int creationYear;
+    private SteeringWheel steeringWheel;
+    private GearShift gearShift;
+    private int currentShift;
+    private double weight;
+    private double consumption;
+
+    /**
+     * Initializing sample car specs for the purpose of the task
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Car.java:24]: The doc doesn't say anything. What is the purpose of the task?
I would rather start with something like: Creates a new Car instance with default values which are...

File: CarApp/src/application/Car.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    public Car() {
+        this.doors = 4;
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Car.java:27]: Extract all fixed values in constants

File: CarApp/src/application/Car.java [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        this.creationYear = 2005;
+        this.steeringWheel = SteeringWheel.LEFT;
+        this.gearShift = GearShift.MANUAL;
+        this.currentShift = 3;
+        this.weight = 3200;
+        this.consumption = 6.6;
+    }
+
+    /**
+     * Simple method printing the specs of the car object on a single line using String.format
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Car.java:39]: Normally you don't include such details in the docs: "using String.format"

File: CarApp/src/application/Car.java [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    public void printCarInfo() {
+        System.out.println((String.format("Car doors: %d, seats: %d, color: %s, year: %d, steering wheel: %s, gear shift: %s, currentShift: %d, weight: %.2fkg, consumption: %.1fl",
+                doors, seats, color, creationYear, steeringWheel, gearShift, currentShift, weight, consumption)));
+    }
+
+    /**
+     * This method checks if the color of the car object is the same
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Car.java:47]: "This method" is not necessary

File: CarApp/src/application/Car.java [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+    /**
+     * Method to check if the capacity of the car is under the maximum, maximum capacity of a single car is 3500kg
+     *
+     * @param bags       count of the bags to be put in the car, a single bag is approx. 40kg
+     * @param passengers count of passengers to get in the car, a single passenger is approx. 60kg
+     * @return returns whether the total weight is smaller than the max capacity of the car
+     */
+    public boolean isWeightUnderMaxWeight(int bags, int passengers) {
+        double x = this.weight + bags * 40 + passengers * 60;
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Car.java:109]: use a more meaningful name, not "x"

File: CarApp/src/application/Enums.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package application;
+
+enum SteeringWheel {
```
Comment (nikolayivanov-deltasource) [inline CarApp/src/application/Enums.java:3]: Each enum in a separate file ;)

**Pull request merged**